src/target.zig

```diff
@@ -459,7 +459,8 @@ pub fn llvmMachineAbi(target: std.Target) ?[:0]const u8 {
     }
 }
 
-/// This function returns 1 if function alignment is not observable or settable.
+/// This function returns 1 if function alignment is not observable or settable. Note that this
+/// value will not necessarily match the backend's default function alignment (e.g. for LLVM).
 pub fn defaultFunctionAlignment(target: std.Target) Alignment {
```

Comment:
What's this function actually used for now? If it doesn't match what the backends do, and it's not the ABI minimum, I don't see what this is for.

Reply:
It's used in a couple of places for self-hosted right now:

https://github.com/ziglang/zig/blob/8573836892ba1b7cd34d377b46258930161256c3/src/codegen.zig#L143-L146

https://github.com/ziglang/zig/blob/8573836892ba1b7cd34d377b46258930161256c3/src/link/Dwarf.zig#L2362-L2378

I'm not entirely sure if that second one is correct because I don't see any matching calls in the self-hosted backends. However, the fix there should be to make the self-hosted backends actually use it, because we do want to align functions to get better performance (except in the case of `-OReleaseSmall` potentially).

Reply:
Ok, went over the remaining uses of both functions. Everything should make more sense now.